src/core/utils/positioningutils.cpp

```diff
@@ -0,0 +1,113 @@
+
+#include "gnsspositioninformation.h"
+#include "positioningutils.h"
+
+PositioningUtils::PositioningUtils( QObject *parent )
+  : QObject( parent )
+{
+}
+
+GnssPositionInformation PositioningUtils::createGnssPositionInformation( double latitude, double longitude, double altitude, double speed, double direction, double horizontalAccuracy, double verticalAcurracy, double verticalSpeed, double magneticVariation, const QDateTime &timestamp, const QString &sourceName )
+{
+  return GnssPositionInformation( latitude, longitude, altitude,
+                                  speed, direction, QList<QgsSatelliteInfo>(), 0, 0, 0,
+                                  horizontalAccuracy, verticalAcurracy, timestamp,
+                                  QChar(), 0, -1, 0, QChar( 'A' ), QList<int>(), false,
+                                  verticalSpeed, magneticVariation, sourceName );
+}
+
+GnssPositionInformation PositioningUtils::averagedPositionInformation( const QList<QVariant> &positionsInformation )
+{
+  QList<GnssPositionInformation> convertedList;
+  for ( const QVariant &pi : positionsInformation )
+  {
+    if ( pi.canConvert<GnssPositionInformation>() )
+    {
+      convertedList << pi.value<GnssPositionInformation>();
+    }
+  }
+  return averagedPositionInformation( convertedList );
+}
+
+GnssPositionInformation PositioningUtils::averagedPositionInformation( const QList<GnssPositionInformation> &positionsInformation )
+{
+  if ( positionsInformation.isEmpty() )
+    return GnssPositionInformation();
+
+  const int positionsInformationCount = positionsInformation.size();
+
+  double latitude = std::numeric_limits<double>::quiet_NaN();
+  double longitude = std::numeric_limits<double>::quiet_NaN();
+  double elevation = std::numeric_limits<double>::quiet_NaN();
+  double speed = std::numeric_limits<double>::quiet_NaN();
+  double direction = std::numeric_limits<double>::quiet_NaN();
+  double pdop = 0;
+  double hdop = 0;
+  double vdop = 0;
+  double hacc = std::numeric_limits<double>::quiet_NaN();
+  double vacc = std::numeric_limits<double>::quiet_NaN();
+  double hvacc = std::numeric_limits<double>::quiet_NaN();
+  double verticalSpeed = std::numeric_limits<double>::quiet_NaN();
+  double magneticVariation = std::numeric_limits<double>::quiet_NaN();
+
+  QDateTime utcDateTime = positionsInformation.last().utcDateTime();
+
+  QList<QgsSatelliteInfo> satellitesInView = positionsInformation.at( 0 ).satellitesInView();
+  int satellitesUsed = satellitesInView.size();
+  bool satInfoComplete = positionsInformation.at( 0 ).satInfoComplete();
+  QList<int> satPrn = positionsInformation.at( 0 ).satPrn();
+  QChar status = positionsInformation.at( 0 ).status();
+  QChar fixMode = positionsInformation.at( 0 ).fixMode();
+  int fixType = positionsInformation.at( 0 ).fixType();
+  int quality = positionsInformation.at( 0 ).quality();
+  QString sourceName = positionsInformation.at( 0 ).sourceName();
```

Comment:
Should we do "sourceName (averaged)" or something like that.
Basically some info somewhere for a proper quality audit trail.

Reply:
I was thinking we could add two new variables (@gnss_averaged and @gnss_averaged_count or something)

Reply:
One would be enough (count=1 is non averaged)

Reply:
@m-kuhn , true.

Reply:
Implemented; value == 0 means non averaged, value >= 1 means averaged.